### DISCUSSION TITLE
setting up rsync to backup from littlebackupbox to Synology NAS

### DISCUSSION THREAD

**Discussion by brentener** [outdoorbits/little-backup-box#210] [General]
Hello !

I have been trying to setup rsync on my Synology NAS (with DSM 7.1.1). When I try to do the backup on my LBB, I get the error code 5. I think this has to do with permissions, but I have no clue what else to change, as I thought I already had granted access to the NAS. I am somehow stuck now. Maybe someone could help me out of the ditch ! Thanks !

**Comment by outdoorbits** [comment]
Did you try to connect to your public address? If yes, did you open your routers port TCP 873 for rsync and redirected this to your Synology?

**Reply by brentener** [reply]
I wanted to try the backup via rsync locally first. After that i can do the settings to setup for external access. I have setup everything in DSM as it should, have grantwd access to external users which i also created in dsm! but as rsync gives me error code 5, there seems to be a permission problem !!!

**Comment by outdoorbits** [comment]
Did you read  readme_rsync.md (lbb)? Did you configure a modul name and configure it correctly in your lbb config?

**Comment by brentener** [comment]
[little-backup-box-3.log](https://github.com/outdoorbits/little-backup-box/files/11851215/little-backup-box-3.log)

I attached the log file from the LBB. 

Here is the rsyncd.conf file:

#global (global settings)
log file = /var/log/rsync.log
timeout = 300
max connections = 0

#profiles
[NetBackup]
path = /var/services/NetBackup
comment = NetBackup
max connections = 2
hosts allow = *
#hosts deny = *
use chroot = yes
list = true
uid = brentener
gid = administrators
read only = false
auth users = brentener
secrets file = /etc/rsyncd.secrets

**Comment by brentener** [comment]
I fixed the rsyncd.conf !!! Now I get the following error:

Backup log:
---- try 1 /media/source ----
2023/06/23 23:06:02 [13866] building file list
2023/06/23 23:06:02 [13866] rsync: [sender] read error: Connection reset by peer (104)
2023/06/23 23:06:02 [13866] sent 12562 bytes  received 0 bytes  total size 1853632448
2023/06/23 23:06:02 [13866] rsync error: error in socket IO (code 10) at io.c(784) [sender=3.2.3]

**Comment by outdoorbits** [comment]
In my installation system users and rsync users are different. Can you create rsync specific users?

**Reply by brentener** [reply]
It is the rsync username I created in DSM under the rsync tab.

**Comment by brentener** [comment]
here are some more lines from the rsync.log file:
2023/06/24 07:07:44 [11512] sent 2235 bytes  received 14800 bytes  total size 1853632448
2023/06/24 07:07:44 [11516] rsync to NetBackup/// from brentener@UNKNOWN (192.168.178.32)
2023/06/24 07:07:44 [11516] rsync: on remote machine: --mkpath: unknown option
2023/06/24 07:07:45 [11516] rsync error: requested action not supported (code 4) at clientserver.c(1713) [Receiver=3.1.2]
2023/06/24 07:07:45 [11519] rsync to NetBackup// from brentener@UNKNOWN (192.168.178.32)
2023/06/24 07:07:45 [11519] receiving file list
2023/06/24 07:07:45 [11519] ./
rsync on LBB seems to create a bad folder path !!!!

**Comment by outdoorbits** [comment]
There is one idea:

uid = brentener
gid = administrators

... is wrong, you need the ID-numbers of uid and gid. Synology seems to print this by the command "id".

**Reply by brentener** [reply]
I changed the uid and pid to 1026 and 101 respectively. After giving the backup to rsync another try, I got error 10 again. Here is the part of the rsync.log when the error occured:

2023/06/25 06:37:09 [17229] sent 2219 bytes  received 14796 bytes  total size 1853632448
2023/06/25 08:37:09 [17234] name lookup failed for 192.168.178.32: Name or service not known
2023/06/25 08:37:09 [17234] connect from UNKNOWN (192.168.178.32)
2023/06/25 06:37:09 [17234] rsync to NetBackup/// from brentener@UNKNOWN (192.168.178.32)
2023/06/25 06:37:09 [17234] rsync: on remote machine: --mkpath: unknown option
2023/06/25 06:37:10 [17234] rsync error: requested action not supported (code 4) at clientserver.c(1713) [Receiver=3.1.2]
2023/06/25 08:37:10 [17238] name lookup failed for 192.168.178.32: Name or service not known
2023/06/25 08:37:10 [17238] connect from UNKNOWN (192.168.178.32)
2023/06/25 06:37:10 [17238] rsync to NetBackup// from brentener@UNKNOWN (192.168.178.32)
2023/06/25 06:37:10 [17238] receiving file list
2023/06/25 06:37:10 [17238] ./

and here is the rsyncd.conf file:

#global (global settings)
log file = /var/log/rsync.log
timeout = 300
max connections = 0

#profiles
[NetBackup]
path = /volume1
comment = NetBackup
max connections = 2
hosts allow = *
#hosts deny = *
use chroot = yes
list = true
uid = 1026
gid = 101
read only = false
auth users = brentener
secrets file = /etc/rsyncd.secrets

**Reply by brentener** [reply]
I think I found another clue: the rsync server version on my DSM 7.1.1 is probably outdated. It is version 3.1.3 and should be 3.2.3 at least !

**Comment by outdoorbits** [comment]
Bad news? Is there a way to upgrade? What's about using nextcloud?

**Reply by brentener** [reply]
I am just about installing a ubuntu server on a virtual machine on my Synology ! From there I can install the latest rsync release ! Finger crossed it's working !
By the way ! I want to use LBB on our 4 weeks trip to Hawaii. That's for I want to be sure to be able to backup my photos to my synology.

**Comment by outdoorbits** [comment]
Great plan! Keep in mind, you can also use Google drive, Dropbox and and and... 
Have a nice trip!
